pyparsing/results.py

```diff
@@ -66,41 +66,41 @@ def test(s, fn=repr):
         - year: 1999
     """
 
+    __slots__ = [
+        "_result__name", "_result__parent", "_result__all_names",
+        "_result__modal", "_result__list", "_result__dict", "__weakref__"
+    ]
+
```

Comment:
We will strictly declare the `__slots__` so that the objects are, hopefully, smaller and uniform.

Reply:
Using `__slots__` also has a memory benefit, which will be of great help in this class.